packages/cli/src/UserManagement/routes/users.ts

```diff
@@ -122,91 +135,81 @@ export function usersNamespace(this: N8nApp): void {
+		ResponseHelper.send(async (req: UserRequest.ResolveSignUp) => {
+			const { inviterId, inviteeId } = req.query;
 
 			if (!inviterId || !inviteeId) {
-				LoggerProxy.error('Invalid invite URL - did not receive user IDs', {
-					inviterId,
-					inviteeId,
-				});
-				throw new ResponseHelper.ResponseError('Invalid payload', undefined, 500);
+				throw new ResponseHelper.ResponseError('Invalid payload', undefined, 400);
 			}
 
 			const users = await Db.collections.User!.find({ where: { id: In([inviterId, inviteeId]) } });
 
 			if (users.length !== 2) {
-				LoggerProxy.error('Invalid invite URL - did not find users', { inviterId, inviteeId });
-				throw new ResponseHelper.ResponseError('Invalid invite URL', undefined, 500);
+				throw new ResponseHelper.ResponseError('Invalid invite URL', undefined, 400);
 			}
 
 			const inviter = users.find((user) => user.id === inviterId);
 
 			if (!inviter || !inviter.email || !inviter.firstName) {
-				LoggerProxy.error('Invalid invite URL - inviter does not have email set', {
-					inviterId,
-					inviteeId,
-				});
-				throw new ResponseHelper.ResponseError('Invalid request', undefined, 500);
+				throw new ResponseHelper.ResponseError('Invalid invite URL', undefined, 400);
 			}
+
 			const { firstName, lastName } = inviter;
 
 			return { inviter: { firstName, lastName } };
 		}),
 	);
 
+	/**
+	 * Fill out user shell with first name, last name, and password.
+	 *
+	 * Authless endpoint.
+	 */
 	this.app.post(
-		`/${this.restEndpoint}/user`,
-		ResponseHelper.send(async (req: AuthenticatedRequest, res: Response) => {
-			if (req.user) {
-				throw new ResponseHelper.ResponseError(
-					'Please logout before accepting another invite.',
-					undefined,
-					500,
-				);
-			}
+		`/${this.restEndpoint}/users/:id`,
+		ResponseHelper.send(async (req: UserRequest.Update, res: Response) => {
+			const { id: inviteeId } = req.params;
 
-			const { inviterId, inviteeId, firstName, lastName, password } = req.body as {
-				inviterId: string;
-				inviteeId: string;
-				firstName: string;
-				lastName: string;
-				password: string;
-			};
+			const { inviterId, firstName, lastName, password } = req.body;
 
 			if (!inviterId || !inviteeId || !firstName || !lastName || !password) {
-				throw new ResponseHelper.ResponseError('Invalid payload', undefined, 500);
+				throw new ResponseHelper.ResponseError('Invalid payload', undefined, 400);
 			}
 
+			const validPassword = validatePassword(password);
+
 			const users = await Db.collections.User!.find({
 				where: { id: In([inviterId, inviteeId]) },
 			});
 
 			if (users.length !== 2) {
-				throw new ResponseHelper.ResponseError('Invalid invite URL', undefined, 500);
+				throw new ResponseHelper.ResponseError('Invalid payload or URL', undefined, 400);
 			}
 
-			const invitee = users.find((user) => user.id === inviteeId);
+			const invitee = users.find((user) => user.id === inviteeId) as User;
```

Comment:
Both options work, but `as User` more clearly reflects intent - we know the return type will be a `User`, rather than anything that is not `null` and not `undefined`.